build.gradle

```diff
@@ -320,13 +328,13 @@ task cleanGenerated(type: Delete, group: 'generate') {
 task generateHaxeParser(dependsOn: ':setupTools', type: JavaExec, group: 'generate') {
     workingDir = "${toolDir}"
     main '-jar'
-    args = ['grammar-kit.jar', "${generatedSrcDir}", "${grammarHaxe}"]
+    args = ['--add-opens=java.base/java.lang.reflect=ALL-UNNAMED', '--add-opens=java.base/java.util=ALL-UNNAMED', '--add-opens=java.base/java.lang=ALL-UNNAMED', 'grammar-kit.jar', "${generatedSrcDir}", "${grammarHaxe}"]
```

Comment:
these and next jvmArgs are added as reflection is restricted on newer JREs

Reply:
> these and next jvmArgs are added as reflection is restricted on newer JREs

@rosingrind beware that GrammarKit has a Gradle plugin now, exposing GenerateLexerTask and GenerateParserTask.

Reply:
totally understandable, however the next step you should consider to perform transition to newer versions - dropping `gradle-grammar-kit-plugin` completely as it does not work as expected, see https://github.com/JetBrains/gradle-grammar-kit-plugin/issues/3

Reply:
I left `gradle-grammar-kit-plugin` configurations as-is because it haven't introduced new problems yet, but it may do so potentially. Correct implementation of utilizing `gradle-grammar-kit-plugin` means double-pass generation as there are some clashing problems in gradle JVM and gk-plugin generation steps, see linked issue for more info

Reply:
@rosingrind interesting. I wasn't aware of this, and I see it's a long standing issue too. Good catch, thanks!

Reply:
i have wanted to upgrade grammar-kit and use the plugin but the reason why we stuck with the old version was because newer versions would output different code on different operating systems. but this could have been solved by now, so we could test the plugin and se how it goes